src/main/webapp/app/shared/list/FormListField.tsx

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import { observer } from 'mobx-react';
+import styles from './styles.module.scss';
+import { Row, Col, Button } from 'react-bootstrap';
+import { AvGroup, AvInput } from 'availity-reactstrap-validation';
+import autobind from 'autobind-decorator';
+import { action, observable } from 'mobx';
+
+type IListBoxProps = {
+  list: string[];
+  onDelete: (item: string) => void;
+};
+
+@observer
+class ListBox extends React.Component<IListBoxProps> {
+  render() {
+    return (
+      <div
+        style={{
+          border: '1px solid #ced4da',
+          borderRadius: '0.25rem',
+          padding: '0.375rem 0.75rem',
+          overflowY: 'scroll',
+          minHeight: '100px',
+          maxHeight: '100px',
+          marginTop: '28px',
+        }}
+      >
+        <div className={styles.main}>
+          {this.props.list.map(item => (
+            <div key={item} className={styles.content_container}>
+              <span
+                className={styles.delete}
+                onClick={() => this.props.onDelete(item)}
+              >
+                <i className="fa fa-times-circle"></i>
+              </span>
+              <span style={{ margin: '5px 10px 5px 5px' }}>{item}</span>
+            </div>
+          ))}
+        </div>
+      </div>
+    );
+  }
+}
+
+type IFormListFieldProps = {
+  list: string[];
+  addItem: (item: string) => void;
+  deleteItem: (item: string) => void;
+  labelText: string;
+  placeholder?: string;
+  boldLabeL?: boolean;
+};
+
+@observer
+export class FormListField extends React.Component<IFormListFieldProps> {
+  @observable isInputValid = false;
+  @observable inputText = '';
+  @observable errorMessage = '';
+
+  @autobind
+  listLengthValidation(value: any, context: any, input: any, cb: any) {
```

Comment:
I think we can type the `cb`?

Reply:
Yeah, the callback is a function that takes a boolean or string and returns void.